<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DlgBrokerProperties</class>
 <widget class="QDialog" name="DlgBrokerProperties">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>603</width>
    <height>733</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Server properties</string>
  </property>
  <layout class="QFormLayout" name="formLayout">
   <item row="0" column="0">
    <widget class="QLabel" name="label">
     <property name="text">
      <string>&amp;Name</string>
     </property>
     <property name="buddy">
      <cstring>edName</cstring>
     </property>
    </widget>
   </item>
   <item row="0" column="1">
    <widget class="QLineEdit" name="edName"/>
   </item>
   <item row="1" column="0">
    <widget class="QLabel" name="label_16">
     <property name="text">
      <string>S&amp;cheme</string>
     </property>
     <property name="buddy">
      <cstring>cbxScheme</cstring>
     </property>
    </widget>
   </item>
   <item row="1" column="1">
    <widget class="QComboBox" name="cbxScheme"/>
   </item>
   <item row="2" column="0">
    <widget class="QLabel" name="label_3">
     <property name="text">
      <string>&amp;Host</string>
     </property>
     <property name="buddy">
      <cstring>edHost</cstring>
     </property>
    </widget>
   </item>
   <item row="2" column="1">
    <widget class="QLineEdit" name="edHost"/>
   </item>
   <item row="3" column="0">
    <widget class="QLabel" name="label_4">
     <property name="text">
      <string>&amp;Port</string>
     </property>
     <property name="buddy">
      <cstring>edPort</cstring>
     </property>
    </widget>
   </item>
   <item row="3" column="1">
    <widget class="QSpinBox" name="edPort">
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
     <property name="minimum">
      <number>256</number>
     </property>
     <property name="maximum">
      <number>65535</number>
     </property>
     <property name="value">
      <number>4840</number>
     </property>
    </widget>
   </item>
   <item row="5" column="0">
    <widget class="QLabel" name="label_14">
     <property name="text">
      <string>&amp;Security type</string>
     </property>
     <property name="buddy">
      <cstring>lstSecurityType</cstring>
     </property>
    </widget>
   </item>
   <item row="5" column="1">
    <widget class="QComboBox" name="lstSecurityType">
     <property name="toolTip">
      <string>Security type</string>
     </property>
    </widget>
   </item>
   <item row="6" column="0">
    <widget class="QLabel" name="label_15">
     <property name="text">
      <string>Peer &amp;verify</string>
     </property>
     <property name="buddy">
      <cstring>chkPeerVerify</cstring>
     </property>
    </widget>
   </item>
   <item row="6" column="1">
    <widget class="QCheckBox" name="chkPeerVerify">
     <property name="layoutDirection">
      <enum>Qt::LeftToRight</enum>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
   </item>
   <item row="7" column="0">
    <widget class="QLabel" name="label_6">
     <property name="text">
      <string>Connection t&amp;ype</string>
     </property>
     <property name="buddy">
      <cstring>cbxConnectionType</cstring>
     </property>
    </widget>
   </item>
   <item row="7" column="1">
    <widget class="QComboBox" name="cbxConnectionType">
     <property name="toolTip">
      <string>Connection type</string>
     </property>
    </widget>
   </item>
   <item row="8" column="0">
    <widget class="QLabel" name="label_7">
     <property name="text">
      <string>Protocol type</string>
     </property>
     <property name="buddy">
      <cstring>rpc_protocolType</cstring>
     </property>
    </widget>
   </item>
   <item row="8" column="1">
    <widget class="QComboBox" name="rpc_protocolType">
     <property name="toolTip">
      <string>Connection type</string>
     </property>
    </widget>
   </item>
   <item row="9" column="0">
    <widget class="QLabel" name="label_13">
     <property name="text">
      <string>SHV &amp;root</string>
     </property>
     <property name="buddy">
      <cstring>shvRoot</cstring>
     </property>
    </widget>
   </item>
   <item row="9" column="1">
    <widget class="QLineEdit" name="shvRoot"/>
   </item>
   <item row="10" column="0">
    <widget class="QLabel" name="label_10">
     <property name="text">
      <string>Reconnect &amp;interval</string>
     </property>
     <property name="buddy">
      <cstring>rpc_reconnectInterval</cstring>
     </property>
    </widget>
   </item>
   <item row="10" column="1">
    <widget class="QSpinBox" name="rpc_reconnectInterval">
     <property name="toolTip">
      <string>Reconnect to broker if connection lost at least after recoonect-interval seconds. Disabled when set to 0</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
     <property name="suffix">
      <string> sec</string>
     </property>
     <property name="maximum">
      <number>999999999</number>
     </property>
     <property name="singleStep">
      <number>10</number>
     </property>
     <property name="value">
      <number>10</number>
     </property>
    </widget>
   </item>
   <item row="11" column="0">
    <widget class="QLabel" name="label_11">
     <property name="text">
      <string>Heart beat interval</string>
     </property>
    </widget>
   </item>
   <item row="11" column="1">
    <widget class="QSpinBox" name="rpc_heartbeatInterval">
     <property name="toolTip">
      <string>Send heart beat to broker every n sec. Disabled when set to 0.</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
     <property name="suffix">
      <string> sec</string>
     </property>
     <property name="maximum">
      <number>999999999</number>
     </property>
     <property name="singleStep">
      <number>10</number>
     </property>
     <property name="value">
      <number>60</number>
     </property>
    </widget>
   </item>
   <item row="12" column="0">
    <widget class="QLabel" name="label_12">
     <property name="text">
      <string>RPC Timeout</string>
     </property>
    </widget>
   </item>
   <item row="12" column="1">
    <widget class="QSpinBox" name="rpc_timeout">
     <property name="toolTip">
      <string>Send heart beat to broker every n sec. Disabled when set to 0.</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
     <property name="suffix">
      <string> sec</string>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>999999999</number>
     </property>
     <property name="singleStep">
      <number>10</number>
     </property>
     <property name="value">
      <number>5</number>
     </property>
    </widget>
   </item>
   <item row="13" column="0" colspan="2">
    <widget class="QGroupBox" name="grpDevice">
     <property name="title">
      <string>Device</string>
     </property>
     <layout class="QFormLayout" name="formLayout_2">
      <item row="1" column="0">
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>Mount point</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="device_mountPoint"/>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>Device ID</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="device_id"/>
      </item>
     </layout>
    </widget>
   </item>
   <item row="14" column="0" colspan="2">
    <widget class="QCheckBox" name="chkMuteHeartBeats">
     <property name="toolTip">
      <string>Mute heart beats notificcations for known types of clients, like Eyas </string>
     </property>
     <property name="text">
      <string>Mute heart beats</string>
     </property>
    </widget>
   </item>
   <item row="15" column="0">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="16" column="0" colspan="2">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QDialogButtonBox" name="buttonBox">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="standardButtons">
        <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="4" column="0" colspan="2">
    <widget class="QGroupBox" name="grpLogin">
     <property name="title">
      <string>Login</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="3">
       <widget class="QCheckBox" name="cbLoginWithAzure">
        <property name="text">
         <string>login with Azure</string>
        </property>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QCheckBox" name="cbPlainTextPassword">
        <property name="text">
         <string>plain text password</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="edUser"/>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Pass&amp;word</string>
        </property>
        <property name="buddy">
         <cstring>edPassword</cstring>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="edPassword">
        <property name="echoMode">
         <enum>QLineEdit::EchoMode::Password</enum>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>&amp;User</string>
        </property>
        <property name="buddy">
         <cstring>edUser</cstring>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QToolButton" name="tbShowPassword">
        <property name="text">
         <string>...</string>
        </property>
        <property name="icon">
         <iconset resource="../shvspy.qrc">
          <normaloff>:/shvspy/images/show.svg</normaloff>:/shvspy/images/show.svg</iconset>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>edName</tabstop>
  <tabstop>cbxScheme</tabstop>
  <tabstop>edHost</tabstop>
  <tabstop>edPort</tabstop>
  <tabstop>lstSecurityType</tabstop>
  <tabstop>chkPeerVerify</tabstop>
  <tabstop>cbxConnectionType</tabstop>
  <tabstop>rpc_protocolType</tabstop>
  <tabstop>shvRoot</tabstop>
  <tabstop>rpc_reconnectInterval</tabstop>
  <tabstop>rpc_heartbeatInterval</tabstop>
  <tabstop>rpc_timeout</tabstop>
  <tabstop>device_id</tabstop>
  <tabstop>device_mountPoint</tabstop>
  <tabstop>chkMuteHeartBeats</tabstop>
 </tabstops>
 <resources>
  <include location="../shvspy.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>DlgBrokerProperties</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>394</x>
     <y>616</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>DlgBrokerProperties</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>394</x>
     <y>616</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
